src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -2197,3 +2222,12 @@ type GetResponseFound<TDocument = unknown> = estypes.GetResponse<TDocument> &
 const isFoundGetResponse = <TDocument = unknown>(
   doc: estypes.GetResponse<TDocument>
 ): doc is GetResponseFound<TDocument> => doc.found;
+
+/**
+ * Response headers check to determine if the response is from Elasticsearch
+ * @param headers Response headers
+ * @returns boolean
+ */
+const isSupportedEsServer = (headers: Record<string, unknown> | null) => {
+  return !!headers && headers['x-elastic-product'] === 'Elasticsearch';
```

Comment:
@mshustov Agreed. However, I wanted to minimize changes to services in this PR so that the risk is lower for backporting to a patch release. The alerting team really needs this workaround as soon as possible.

Reply:
> We can revisit in the follow up PR that implements the checks in other SO repository methods.

Could you add a comment to move it to ES service? 

> I'm also not sure exactly where to move it to 🤔

in a dedicated file is okay

Reply:
I'm making other changes to the PR and have moved the method, as requested.